ksqldb-functional-tests/src/test/resources/query-validation-tests/group-by.json

```diff
@@ -572,6 +572,83 @@
         ]
       }
     },
+    {
+      "name": "multiple expressions - KAFKA to struct key format",
+      "statements": [
+        "CREATE STREAM TEST (f1 INT KEY, f2 INT) WITH (kafka_topic='test_topic', value_format='JSON', partitions=1);",
+        "CREATE TABLE OUTPUT WITH (KEY_FORMAT='JSON') AS SELECT STRUCT(f1:=f1, f2:=f2) AS k, as_value(struct(f1:=f1, f2:=f2)) AS key, COUNT(*) AS total FROM TEST GROUP BY STRUCT(f1:=f1, f2:=f2);"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 1, "value": {"f2": "2"}},
+        {"topic": "test_topic", "key": 2, "value": {"f2": "4"}},
+        {"topic": "test_topic", "key": 1, "value": {"f2": null}},
+        {"topic": "test_topic", "key": 1, "value": null},
+        {"topic": "test_topic", "key": null, "value": {"f2": "1"}},
+        {"topic": "test_topic", "key": 1, "value": {"f2": "2"}},
+        {"topic": "test_topic", "key": 2, "value": {"f2": "4"}},
+        {"topic": "test_topic", "key": 2, "value": {"f2": "1"}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": {"F1": 1, "F2": 2}, "value": {"KEY": {"F1":1,"F2":2}, "TOTAL": 1}},
+        {"topic": "OUTPUT", "key": {"F1": 2, "F2": 4}, "value": {"KEY": {"F1":2,"F2":4}, "TOTAL": 1}},
+        {"topic": "OUTPUT", "key": {"F1": 1, "F2": null}, "value": {"KEY": {"F1":1,"F2":null}, "TOTAL": 1}},
+        {"topic": "OUTPUT", "key": {"F1": null, "F2": 1}, "value": {"KEY": {"F1":null,"F2":1}, "TOTAL": 1}},
+        {"topic": "OUTPUT", "key": {"F1": 1, "F2": 2}, "value": {"KEY": {"F1":1,"F2":2}, "TOTAL": 2}},
+        {"topic": "OUTPUT", "key": {"F1": 2, "F2": 4}, "value": {"KEY": {"F1":2,"F2":4}, "TOTAL": 2}},
+        {"topic": "OUTPUT", "key": {"F1": 2, "F2": 1}, "value": {"KEY": {"F1":2,"F2":1}, "TOTAL": 1}}
+      ]
+    },
+    {
+      "name": "multiple expressions - delimited key format to multi-column primitive format",
```

Comment:
What is the purpose of adding this test? In the context of this PR, we should add a test to verify that no key format change is required in this case -- this test should pass without the `(KEY_FORMAT='JSON')` in the CTAS statement.

Reply:
The original bug statement does contain `WITH (KEY_FORMAT='JSON')`. Without it, the original test would also fail, see `multiple expressions - KAFKA to non-KAFKA key format` and remove the `WITH (KEY_FORMAT='JSON')` to reproduce.

I think we are inherently not converting the source format for key construction, but let me check.

Reply:
> The original bug statement does contain WITH (KEY_FORMAT='JSON'). Without it, the existing test would also fail

I agree that the test in the original bug statement fails without `WITH (KEY_FORMAT='JSON')` in the CTAS statement, but for this particular test (delimited key format to multi-column primitive format), the test should not fail if `WITH (KEY_FORMAT='JSON')` is removed, since the sink schema is supported by the delimited format. It would be good to verify that this test still passes without the `WITH (KEY_FORMAT='JSON')`. I also don't think we need to test the version with `WITH (KEY_FORMAT='JSON')` (for this test) since that would no longer be testing the bug in this PR but simply testing the functionality of converting key formats in general, which should already be tested elsewhere.